tests/foreman/cli/test_ansible.py

```diff
@@ -561,3 +566,112 @@ def test_positive_install_ansible_collection(
         assert result['success'] == '1'
         collection_path = client.execute('ls ~/ansible_collections').stdout
         assert 'oasis_roles' in collection_path
+
+    @pytest.mark.tier3
+    @pytest.mark.no_containers
+    @pytest.mark.rhel_ver_list([settings.content_host.default_rhel_version])
+    @pytest.mark.parametrize('auth_type', ['admin', 'non-admin'])
+    def test_positive_ansible_variables_imported_with_roles(
+        self, request, auth_type, target_sat, module_org, module_ak_with_cv, rhel_contenthost
+    ):
+        """Verify that when Ansible roles are imported, their variables are imported simultaneously
+
+        :id: 107c53e8-5a8a-4291-bbde-fbd66a0bb85e
+
+        :steps:
+            1. Register a content host with satellite
+            2. Create a custom role with variables and import it into satellite
+            3. Assert that the role is imported along with associated variables
+            4. Assign that role to a host and verify the role is assigned to the host
+            5. Run the Ansible role
+
+        :expectedresults: Verify variables associated to role are also imported along with roles
```

Comment:
It looks like only steps 2 and 3 are relevant to this test. Why are steps 1, 4, and 5 included? Shouldn't those be part of a separate test, such as verifying that an Ansible role can be assigned to a host?

Reply:
@nofaralfasi You're right, but I've added those steps to make this test more efficient with an end-to-end scenario with custom roles

Reply:
So we don't have another test like that? Where we verify that an Ansible role can be assigned to a host? Or that we can run Ansible job on a host with roles?

Reply:
yes, we have tests covering role assigning and running on host in API, but this test is covering custom role and its an e2e scenario for CLI

Reply:
I agree with @Gauravtalreja1 , We don't need a separate test here. This would avoid duplication and keep the coverage as expected.